src/can.cpp

```diff
@@ -0,0 +1,92 @@
+#include "cand/can.hpp"
+
+CAN::CAN(std::string name)
+    : m_interfaceName { name }
+{
+
+    int socketCan;
+    struct sockaddr_can addr;
+    struct ifreq ifr;
+
+    if ((socketCan = socket(PF_CAN, SOCK_RAW, CAN_RAW)) < 0) {
+        throw std::runtime_error("Socket initialization error");
+    }
+
+    strcpy(ifr.ifr_name, m_interfaceName.c_str());
+    ioctl(socketCan, SIOCGIFINDEX, &ifr);
+
+    std::memset(&addr, 0, sizeof(addr));
+    addr.can_family = AF_CAN;
+    addr.can_ifindex = ifr.ifr_ifindex;
+
+    if (bind(socketCan, (struct sockaddr*)&addr, sizeof(addr)) < 0) {
+        throw std::runtime_error("Binding error");
+    }
+
+    m_fileDescriptor = socketCan;
+}
+
+CAN::~CAN()
+{
+    if (close(m_fileDescriptor) < 0) {
+        throw std::runtime_error("Closing error");
+    }
+}
+
+int CAN::getFileDescriptor()
```

Comment:
Esse método pode ser indicado como const.
```suggestion
auto CAN::getFileDescriptor() const -> int;
```

Reply:
Isso se aplica ao header também